src/benchmarks/micro/corefx/System.Threading.Tasks.Dataflow/Perf.Dataflow.cs

```diff
@@ -0,0 +1,218 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Threading.Tasks.Dataflow.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX)]
+    public abstract class PerfTests<T> where T : IDataflowBlock
+    {
+        protected const int MessagesCount = 100_000;
+        protected T block;
+
+        public abstract T CreateBlock();
+
+        [GlobalSetup]
+        public void BlockSetup()
+        {
+            block = CreateBlock();
+        }
+
+        [Benchmark]
+        public async Task Completion()
+        {
+            BlockSetup();
+            block.Complete();
+            await block.Completion;
+        }
+
+        protected static Task Post(ITargetBlock<int> target, bool retry = false) => Task.Run(() =>
+        {
+            for (int i = 0; i < MessagesCount; i++)
+            {
+                while (!target.Post(i) && retry) ;
+            }
+        });
+
+        protected static Task Receive<U>(ISourceBlock<U> source, int receiveSize = 1) => Task.Run(() =>
+        {
+            for (int i = 0; i < MessagesCount / receiveSize; i++)
+            {
+                source.Receive();
+            }
+        });
+
+        protected static Task TryReceive<U>(IReceivableSourceBlock<U> source, int receiveSize = 1, bool retry = false) => Task.Run(() =>
```

Comment:
Same comment as below for all of these.  The extra task seems superfluous, or if it's necessary for some reason, it should be done by the caller where it's necessary.

Reply:
Good remark! Fixed.